src/fetchRecipe.ts

```diff
@@ -0,0 +1,34 @@
+const baseUrl = `${process.env.REACT_APP_API_URL}`;
+if (!baseUrl || baseUrl === "undefined") {
+  throw new Error("The URL environment variable is undefined/missing");
+}
+export async function FetchRecipe<Response>(id?: string) {
+  if (id) {
+    var url = baseUrl + "/recipe/" + id;
+  } else {
+    var url = baseUrl + "/recipe";
+  }
+
+  const response = await fetch(url);
+  const data: Response = await response.json();
+  if (!data) {
+    throw new Error(
+      "The fetch call did not return any data matching the Recipe Type"
+    );
+  }
+
+  return data;
+}
+
+export async function PostRecipe(
+  options:
+    | { method: string; headers: { "Content-Type": string }; body: string }
+    | undefined
+) {
+  if (!options) {
```

Comment:
Since this code is called by your code, you don't need a null check here.  Just make `options` a required parameter and TS wont let you call it with a possibly undefined object